lib/more_core_extensions/core_ext/shared/nested.rb

```diff
@@ -62,21 +62,25 @@ def has_key_path?(*args)
       def store_path(*args)
         raise ArgumentError, "must pass at least one key, and a value" if args.length < 2
         value = args.pop
-        args = args.first if args.length == 1 && args.first.kind_of?(Array)
+        child = self
+        key   = args.first
 
-        key = args.first
-        raise ArgumentError, "must be a number" if self.kind_of?(Array) && !key.kind_of?(Numeric)
+        if args.length > 1 || args.first.kind_of?(Array)
+          keys = args.first.kind_of?(Array) ? args.first : args
+          keys.each_with_index do |cur_key, i|
+            raise ArgumentError, "must be a number" if kind_of?(Array) && !cur_key.kind_of?(Numeric)
```

Comment:
Just noticed this while writing some specs for mixed groups, and this needs to be `child.kind_of?(Arrray)` or it won't function properly.